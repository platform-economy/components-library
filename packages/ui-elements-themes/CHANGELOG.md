# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
