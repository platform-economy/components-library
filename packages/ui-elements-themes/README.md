# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
