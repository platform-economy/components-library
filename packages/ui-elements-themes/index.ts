export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
