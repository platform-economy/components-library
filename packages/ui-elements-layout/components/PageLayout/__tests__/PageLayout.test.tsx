import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';

import { withTestTheme } from '@relayr/ui-elements-themes';

// Component
import { PageLayout, widthSwitch, paddingSwitch, orientationSwitch } from '../index';

describe('ui-elements-layout', () => {
  describe('Components', () => {
    describe('PageLayout', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(withTestTheme(<PageLayout />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('changes width when mobile property changes', () => {
        const tree = TestRenderer
        .create(withTestTheme(<PageLayout mobile={true} />))
        .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('switches certain styling properties when mobile props is true', () => {
        const props = { mobile: true };

        expect(widthSwitch(props.mobile)).toBe('100%');
        expect(orientationSwitch(props.mobile)).toBe('column');
        expect(paddingSwitch(props.mobile)).toBe('14px 22px');
      });
      it('switches certain styling properties when mobile props is false', () => {
        const props = { mobile: false };

        expect(widthSwitch(props.mobile)).toBe('auto');
        expect(orientationSwitch(props.mobile)).toBe('row');
        expect(paddingSwitch(props.mobile)).toBe('14px 40px');
      });
    });
  });
});
