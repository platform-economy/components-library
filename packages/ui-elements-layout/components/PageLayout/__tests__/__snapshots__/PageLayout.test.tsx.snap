// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-layout Components PageLayout changes width when mobile property changes 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #background;
  min-height: 100vh;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
  background-color: grey;
}

.c3 {
  width: 100%;
}

.c4 {
  padding: 14px 22px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  width: 100%;
  background-color: #divider;
}

<div
  className="Layout c0"
>
  <header
    className="PageHeader c1"
  />
  <div
    className="Main c2"
  >
    <nav
      className="Navigation c3"
    />
    <div
      className="Content c4"
    />
  </div>
  <div
    className="PageFooter c5"
  />
</div>
`;

exports[`ui-elements-layout Components PageLayout should render with default state 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #background;
  min-height: 100vh;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  width: 100%;
  background-color: grey;
}

.c3 {
  width: auto;
}

.c4 {
  padding: 14px 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c5 {
  width: 100%;
  background-color: #divider;
}

<div
  className="Layout c0"
>
  <header
    className="PageHeader c1"
  />
  <div
    className="Main c2"
  >
    <nav
      className="Navigation c3"
    />
    <div
      className="Content c4"
    />
  </div>
  <div
    className="PageFooter c5"
  />
</div>
`;
